pkg/statsutil/stats_linux.go

```diff
@@ -114,18 +119,21 @@ func getHostMemLimit() float64 {
 	return float64(^uint64(0))
 }
 
-func calculateCgroupCPUPercent(previousStats *ContainerStats, metrics *v1.Metrics) float64 {
+func calculateCgroupCPUPercent(previousStats *ContainerStats, metrics *v1.Metrics, systemInfo SystemInfo) float64 {
 	var (
 		cpuPercent = 0.0
 		// calculate the change for the cpu usage of the container in between readings
 		cpuDelta = float64(metrics.CPU.Usage.Total) - float64(previousStats.CgroupCPU)
 		// calculate the change for the entire system between readings
-		systemDelta = float64(metrics.CPU.Usage.Kernel) - float64(previousStats.CgroupSystem)
-		onlineCPUs  = float64(len(metrics.CPU.Usage.PerCPU))
+		systemDelta = float64(systemInfo.SystemUsage) - float64(previousStats.CgroupSystem)
+		onlineCPUs  = systemInfo.OnlineCPUs
 	)
 
+	if onlineCPUs == 0 {
+		onlineCPUs = uint32(len(metrics.CPU.Usage.PerCPU))
+	}
 	if systemDelta > 0.0 && cpuDelta > 0.0 {
-		cpuPercent = (cpuDelta / systemDelta) * onlineCPUs * 100.0
+		cpuPercent = (cpuDelta / systemDelta) * float64(systemInfo.OnlineCPUs) * 100.0
```

Comment:
There's a potential inconsistency in the `calculateCgroupCPUPercent` function where `onlineCPUs` is used in two places:

````
if onlineCPUs == 0 {
    onlineCPUs = uint32(len(metrics.CPU.Usage.PerCPU))
}
// ...
cpuPercent = (cpuDelta / systemDelta) * float64(systemInfo.OnlineCPUs) * 100.0
`````
The second usage directly accesses `systemInfo.OnlineCPUs` instead of using the local onlineCPUs variable that might have been updated. This could lead to inconsistent behavior if the fallback was triggered.